packages/proxy/lib/http/util/top-simulation.ts

```diff
@@ -0,0 +1,31 @@
+import { urlOriginsMatch, urlSameSiteMatch } from '@packages/network/lib/cors'
+import type { HttpMiddlewareThis } from '../index'
+import type { SecFetchSite } from '../../types'
+
+export const doesTopNeedToBeSimulated = <T>(ctx: HttpMiddlewareThis<T>): boolean => {
+  const currentAUTUrl = ctx.getAUTUrl()
+
+  // if the AUT url is undefined for whatever reason, return false as we do not want to complicate top simulation
+  if (!currentAUTUrl) {
+    return false
+  }
+
+  // only simulate top if the AUT is NOT the primary origin, meaning that we should treat the AUT as top
+  // or the request is the AUT frame, which is common for redirects and navigations.
+  const doesTopNeedToSimulating = !ctx.remoteStates.isPrimaryOrigin(currentAUTUrl) || ctx.req.isAUTFrame
+
+  return doesTopNeedToSimulating
+}
+
+// @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site
```

Comment:
this isn't supported in safari? how does will this impact webkit support?

Reply:
So the good news here is we don't actually do anything with the `Sec-Fetch-*` metadata headers here. The reference here is mainly to serve as a a definition point between `same-origin`, `same-site`, and `cross-site`, which I can't seem to find anything that defines this better, even though it isn't related to the metadata headers at all. I will revise this comment to have some context, and likely include a link to [third party cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#third-party_cookies)

Reply:
updated the comments for the method [here](https://github.com/cypress-io/cypress/pull/23880/files#diff-8bcc53731d8dc8219a282445e0ae5ae47e2884d0fb28ead3eef2b488819d1bf0R69). Not sure if this reads a better but should hopefully shed light on the util